### PR TITLE
ITE: drivers/bbram: add magic number to compare in initial

### PR DESCRIPTION
**Pull request by GTLin08**
BBRAM content should be cleared when boot from cutoff. Compare magic
number to decide whether to clear entire BBRAM.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

### CODE REVIEW

File: drivers/bbram/bbram_it8xxx2.c [hunk 1]
```diff
@@ -53,7 +59,30 @@ static const struct bbram_driver_api bbram_it8xxx2_driver_api = {
 
 static int bbram_it8xxx2_init(const struct device *dev)
 {
-	ARG_UNUSED(dev);
+	const struct bbram_it8xxx2_config *config = dev->config;
+	volatile uint8_t *base_addr = (uint8_t *)config->base_addr;
+	volatile uint8_t *bram_valid_flag0 = base_addr + BRAM_IDX_VALID_FLAGS0;
+	volatile uint8_t *bram_valid_flag1 = base_addr + BRAM_IDX_VALID_FLAGS1;
+	volatile uint8_t *bram_valid_flag2 = base_addr + BRAM_IDX_VALID_FLAGS2;
+	volatile uint8_t *bram_valid_flag3 = base_addr + BRAM_IDX_VALID_FLAGS3;
```
Comment (brockus-zephyr) [inline drivers/bbram/bbram_it8xxx2.c:67]: Why is volatile needed for these pointers?
    config->base_addr should not change and therefore removing optimization on base_addr for no reason.
    In this code, there is no redundant accesses to optimize out multiple reads that needs to be managed.

File: soc/riscv/riscv-ite/common/chip_chipregs.h [hunk 2]
```diff
@@ -1869,6 +1869,24 @@ struct gctrl_it8xxx2_regs {
...
+/* Battery backed RAM indices. */
+enum bram_indices {
+
+	/* This field is used to indicate BRAM is valid or not. */
+	BRAM_IDX_VALID_FLAGS0  = 0xbc,
+	BRAM_IDX_VALID_FLAGS1  = 0xbd,
+	BRAM_IDX_VALID_FLAGS2  = 0xbe,
+	BRAM_IDX_VALID_FLAGS3  = 0xbf
+
+};
```
Comment (brockus-zephyr) [inline soc/riscv/riscv-ite/common/chip_chipregs.h:1887]: I would prefer to remove the magic numbers in the code.

#define BRAM_MAGIC_FIELD_OFFSET 0xbc
enum bram_indices {
        /* This field is used to indicate BRAM is valid or not. */
        BRAM_IDX_VALID_FLAGS0  = BRAM_MAGIC_FIELD_OFFSET,
        BRAM_IDX_VALID_FLAGS1,
        BRAM_IDX_VALID_FLAGS2,
        BRAM_IDX_VALID_FLAGS3
};

**Review by brockus-zephyr — CHANGES_REQUESTED** [review]

File: drivers/bbram/bbram_it8xxx2.c (hunk 1 above)
Comment (GTLin08) [reply drivers/bbram/bbram_it8xxx2.c]: Done

File: soc/riscv/riscv-ite/common/chip_chipregs.h (hunk 2 above)
Comment (GTLin08) [reply soc/riscv/riscv-ite/common/chip_chipregs.h:1887]: Done.

**Comment by GTLin08** [comment]
@brockus-zephyr add a new commit for cleanup in this PR.

**Review by keith-zephyr — APPROVED** [review]

**Review by brockus-zephyr — APPROVED** [review]

**Pull request merged**